### PR TITLE
Defensive code to run optimizers

### PR DESCRIPTION
**Pull request by caeugusmao**

### CODE REVIEW

File: thumbor/optimizers/gifv.py [hunk 1]
```diff
@@ -15,20 +15,28 @@
...
 
 
 class Optimizer(BaseOptimizer):
 
     def should_run(self, image_extension, buffer):
-        return 'gif' in image_extension and 'gifv' in self.context.request.filters
+        if 'gif' in image_extension and 'gifv' in self.context.request.filters:
+            if not exists(self.context.config.FFMPEG_PATH):
+                logger.warn(
+                    'gifv optimize enabled but binary JPEGTRAN_PATH does not exist')
```
Comment (heynemann) [inline thumbor/optimizers/gifv.py]: Did you mean FFMPEG_PATH?

Comment (heynemann) [reply thumbor/optimizers/gifv.py]: Also gifv optimizeR, instead of optimize.

File: thumbor/optimizers/jpegtran.py [hunk 2]
```diff
@@ -12,12 +12,19 @@
...
 
 
 class Optimizer(BaseOptimizer):
 
     def should_run(self, image_extension, buffer):
-        return image_extension in ['.jpg', '.jpeg']
+        if image_extension in ['.jpg', '.jpeg']:
+            if not exists(self.context.config.JPEGTRAN_PATH):
+                logger.warn(
+                    'jpegtran optimize enabled but binary JPEGTRAN_PATH does not exist')
```
Comment (heynemann) [inline thumbor/optimizers/jpegtran.py]: jpegtran optimizeR, not optimize.

File: thumbor/optimizers/gifv.py (hunk 1 above)
Comment (caeugusmao) [reply thumbor/optimizers/gifv.py]: Fixed!!! Sorry, my mistake!!!

File: thumbor/optimizers/jpegtran.py (hunk 2 above)
Comment (caeugusmao) [reply thumbor/optimizers/jpegtran.py]: Fixed!!! Sorry, my mistake!!!

**Comment by heynemann** [comment]
Perfect! Thanks!

**Comment by scorphus** [comment]
Nice! I'm glad to see contributions from you, @caeugusmao!

Don't forget to add your email address to your account settings by following the steps in ["Adding an email address to your GitHub account"](https://help.github.com/articles/adding-an-email-address-to-your-github-account/). That way, [these contributions](/thumbor/thumbor/commit/0f5e5d98f699440180803960b3159cce935e3f9a) will be linked to [your account](/caeugusmao).

**Pull request merged**